File: pythonTrack/week1/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
       
       vector <int> num;
       int diff;
       unordered_map<int, int> m;
       for(int i=0;i<nums.size();i++){
            diff=target-nums[i];
            if(m.find(diff)!=m.end())
       return {m[diff],i};
   m[nums[i]] = i;

       }
    //     for(int i=0;i<nums.size();i++){
    //         for(int j=i+1;j<nums.size();j++){
    //         if((nums[i]+nums[j])==target){
    //    num.push_back(i);
    //     num.push_back(j);
    //      return num;}
         
    //     }
    //     }
    return num;
    }
};

File: pythonTrack/week2/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        res = []
        l = 0
        r = len(nums) - 1
        for i in range(len (nums)):
            for j in range (i+1,len(nums)):
                s=nums[i]+nums[j]
                if s==target:
                    res.append(i)
                    res.append(j)
                    return res
                
        return res